src/index.ts

```diff
@@ -171,3 +171,19 @@ export function memo<T, P extends keyof T>(
   }
   return result;
 }
+
+/**
+ * Validate if a value is an object
```

Comment:
I don't think we should use the word "validate" for this. The reason being that there should eventually be another function added here called validate object, which would raise an exception on validation failure.

We are not doing that with this function, so how about "detect" or "check"?

Reply:
I think check sounds about right.